pkg/server/server.go

```diff
@@ -1379,14 +1461,57 @@ func (s *Server) ListStores(ctx context.Context, req *openfgav1.ListStoresReques
 
 	ctx = telemetry.ContextWithRPCInfo(ctx, telemetry.RPCInfo{
 		Service: s.serviceName,
-		Method:  "ListStores",
+		Method:  authz.ListStores,
 	})
 
+	stores, err := s.checkAuthzListStores(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	q := commands.NewListStoresQuery(s.datastore,
 		commands.WithListStoresQueryLogger(s.logger),
 		commands.WithListStoresQueryEncoder(s.encoder),
 	)
-	return q.Execute(ctx, req)
+
+	if s.authorizer == nil {
+		return q.Execute(ctx, req)
+	}
+
+	storesMap := make(map[string]struct{})
+	re := regexp.MustCompile(`^.*:(.*)`)
+	for _, store := range stores {
+		id := re.FindStringSubmatch(store)
+		storesMap[id[1]] = struct{}{}
+	}
+
+	var accessibleStores []*openfgav1.Store
+	var continuationToken string
+
+	for {
+		req.ContinuationToken = continuationToken
+		resp, err := q.Execute(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, store := range resp.GetStores() {
+			if _, ok := storesMap[store.GetId()]; ok {
```

Comment:
Having business logic in `server.go` breaks the existing pattern of encapsulating the APIs on `command` objects that can be re-used, it makes unit testing harder (because you need a server object), bloats `server.go`, etc.

At the very least, could we put all this logic in a new `ListObjectsWithAuthz` command?

Or, if you feel like it, could you experiment with this idea?

1. Change `NewListStoresQuery` to receive a `Authorizer` (interface, not struct) object that defaults to NoOpAuthorizer
2. Change the `Execute` method so that if `authorizer == NoOp`, then we use the existing code, else we use the code you wrote here.

cc @openfga/backend

Reply:
Temporarily removed ListStores to be addressed in a future PR